minheap-internal.h

```diff
@@ -138,6 +138,10 @@ int min_heap_reserve_(min_heap_t* s, unsigned n)
 		unsigned a = s->a ? s->a * 2 : 8;
 		if (a < n)
 			a = n;
+#if (SIZE_MAX == UINT32_MAX)
+		if (a > SIZE_MAX / sizeof *p)
```

Comment:
> On 32 bit systems with tightly packed memory layout

What is "tightly packed memory layout" exactly means here?

>and adding the same pointer again and again

How is this possible?

Reply:
"Tightly packed" in this context means that you need a system that is able to realloc 1 GB memory to 2 GB. On Linux systems with glibc that is impossible due to memory fragmentation. So it already takes an unusual setup to get a heap of 2 GB in size (which is needed to overflow s->a * 2 in the next min_heap_reserve_ call.

You can trigger this stupid behaviour (of adding a pointer again and again) through public API, namely event_add_nolock_ calling event_queue_insert_timeout calling min_heap_push_.

Definitely an unusual use case and an error on caller side. So we talk about a protection against an overflow on rare 32 bit systems with problematic call traces.

Reply:
>You can trigger this stupid behaviour (of adding a pointer again and again) through public API, namely event_add_nolock_ calling event_queue_insert_timeout calling min_heap_push_.

Hm, I guess you mean `event_add()` (but anyway differs only with lock acquire/release from `event_add_nolock_()`), but it should not add the same timer again and again it should replace it